app/services/puppet_class_importer.rb

```diff
@@ -182,6 +182,10 @@ def old_environments
     db_environments - actual_environments
   end
 
+  def current_user_environments
+    Environment.where(:id => Organization.where(:id => User.current.organization_ids).map { |org| org.environment_ids }.reduce(:+)).map(&:to_s)
```

Comment:
Please don't use `User.current.organization_ids` directly, use [Organization.my_organizations](https://github.com/theforeman/foreman/blob/develop/app/models/taxonomies/organization.rb#L18) instead, it will respect admin exception and nesting of Organizations.